arches_her/media/js/views/components/reports/consultation.js

```diff
@@ -14,31 +14,250 @@ define([
+            };
+
+            const assessmentOfSignificanceNode = self.getRawNodeValue(self.resource(), 'assessment of significance');
+            if(Array.isArray(assessmentOfSignificanceNode)){
+                self.assessmentOfSignificance(assessmentOfSignificanceNode.map(node => {
+                    const notes = self.getNodeValue(node, 'notes');
+                    const tileid = self.getTileId(node);
+                    return {notes, tileid};
+                }));
+            };
+
+            const communicationsNode = self.getRawNodeValue(self.resource(), 'communications');
+            if(Array.isArray(communicationsNode)){
+                self.communications(communicationsNode.map(node => {
+                    const subject = self.getNodeValue(node, 'subjects', 'subject');
+                    const subjectDescType = self.getNodeValue(node, 'subjects', 'subject description type');
+                    const type = self.getNodeValue(node, 'communication type');
+                    const date = self.getNodeValue(node, 'dates', 'date');
+                    // const endDate = self.getNodeValue(node, 'dates', 'end date');
+                    const attendees = self.getNodeValue(node, 'attendees');
+                    const relatedCondition = self.getNodeValue(node, 'related condition');
+                    const note = self.getNodeValue(node, 'communication notes', 'communication description');
+                    // const noteDescType = self.getNodeValue(node, 'communication notes', 'communication description type');
+                    const followOnAction = self.getNodeValue(node, 'follow on actions', 'follow-on actions');
+                    // const followOnActionDescType = self.getNodeValue(node, 'follow on actions', 'follow-on actions description type');
+                    const digitalFile = self.getNodeValue(node, 'digital file(s)');
+                    const digitalFileLink = self.getResourceLink(self.getRawNodeValue(node, 'digital file(s)'));
+                    const tileid = self.getTileId(node);
+                    return {subject, subjectDescType, type, date, attendees, note, followOnAction, relatedCondition, digitalFile, digitalFileLink, tileid};
+                }));
+            };
+
+            const siteVisitsNode = self.getRawNodeValue(self.resource(), 'site visits');
+            if(Array.isArray(siteVisitsNode)){
+                self.siteVisits(siteVisitsNode.map(node => {
+                    const dateOfVisit = self.getNodeValue(node, 'timespan of visit', 'date of visit');
+                    const location = self.getNodeValue(node, 'location', 'location descriptions', 'location description');
+                    const locationDescType = self.getNodeValue(node, 'location', 'location descriptions', 'location description type');
+                    const attendees = self.getRawNodeValue(node, 'attendees').map(attendeeNode => {
```

Comment:
This causes an error whenever the attendees node isn't filled in.  You should check to see if the raw node value is an array first like you do with the siteVisitsNode.  The same is true for the rest of these child nodes.

Reply:
Updated, thanks.